packages/runed/src/lib/utilities/element-size/element-size.svelte.ts

```diff
@@ -22,50 +23,77 @@ export type ElementSizeOptions = ConfigurableWindow & {
  * @see {@link https://runed.dev/docs/utilities/element-size}
  */
 export class ElementSize {
-	#size = $state({
+	// no need to use `$state` here since we are using createSubscriber
+	#size = {
 		width: 0,
 		height: 0,
-	});
+	};
 
-	constructor(node: MaybeElementGetter, options: ElementSizeOptions = { box: "border-box" }) {
-		const window = options.window ?? defaultWindow;
+	#options: ElementSizeOptions;
 
-		this.#size = {
-			width: options.initialSize?.width ?? 0,
-			height: options.initialSize?.height ?? 0,
-		};
+	#node: MaybeElementGetter;
+
+	#window: ElementSizeOptions["window"];
 
-		$effect(() => {
-			if (!window) return;
-			const node$ = get(node);
-			if (!node$) return;
+	// we use a derived here to extract the width so that if the width doesn't change we don't get a state update
+	// which we would get if we would just use a getter since the version of the subscriber will be changing
+	#width = $derived.by(() => {
+		this.#subscribe?.();
```

Comment:
Can we add the subscribe calll inside the get width method instead? Same with height

Reply:
Nope because when you call subscribe you register a dependency on the version of the subscriber, since that's incremented when you call update if you add the subscribe in the getter it will rerun even if the height is the same